lib/fetch/headers.js

```diff
@@ -309,23 +290,25 @@ class Headers {
   }
 
   * keys () {
-    const clone = this[kHeadersList].slice()
-    for (let index = 0; index < clone.length; index += 2) {
-      yield clone[index]
+    console.log('called')
+    const listFreezed = new HeadersList(this[kHeadersList])
+    for (const key of listFreezed.keys()) {
+      yield key
     }
   }
 
   * values () {
-    const clone = this[kHeadersList].slice()
-    for (let index = 1; index < clone.length; index += 2) {
-      yield clone[index]
+    const listFreezed = new HeadersList(this[kHeadersList])
+    for (const val of listFreezed.values()) {
+      yield val
     }
   }
 
   * entries () {
-    const clone = this[kHeadersList].slice()
-    for (let index = 0; index < clone.length; index += 2) {
-      yield [clone[index], clone[index + 1]]
+    console.log('called')
+    const listFreezed = new HeadersList(this[kHeadersList])
```

Comment:
I had to create a frozen list because the `Map` iterator doesn't freeze the values/keys.